be/src/Project/Web/Buyer/Buyer.Web/Areas/Orders/ApiControllers/BasketsApiController.cs

```diff
@@ -50,31 +54,21 @@ public async Task<IActionResult> Index([FromBody] SaveBasketRequestModel model)
             var token = await HttpContext.GetTokenAsync(ApiExtensionsConstants.TokenName);
             var language = CultureInfo.CurrentUICulture.Name;
 
-            var reqCookie = this.Request.Cookies[BasketConstants.BasketCookieName];
+            var reqCookie = Request.Cookies[BasketConstants.BasketCookieName];
             if (reqCookie is null)
             {
                 reqCookie = Guid.NewGuid().ToString();
                 var cookieOptions = new CookieOptions
                 {
                     MaxAge = TimeSpan.FromDays(BasketConstants.BasketCookieMaxAge)
                 };
-                this.Response.Cookies.Append(BasketConstants.BasketCookieName, reqCookie, cookieOptions);
+                Response.Cookies.Append(BasketConstants.BasketCookieName, reqCookie, cookieOptions);
             }
 
+            var items = await _basketItemsService.GetBasketItemsAsync(token, language, model.Items);
+
             var id = Guid.Parse(reqCookie);
-            var basket = await this.basketRepository.SaveAsync(token, language, id,
-                model.Items.OrEmptyIfNull().Select(x => new BasketItem
-                {
-                    ProductId = x.ProductId,
-                    ProductSku = x.Sku,
-                    ProductName = x.Name,
-                    PictureUrl = !string.IsNullOrWhiteSpace(x.ImageSrc) ? x.ImageSrc : (x.ImageId.HasValue ? this.mediaService.GetMediaUrl(x.ImageId.Value, OrdersConstants.Basket.BasketProductImageMaxWidth) : null),
-                    Quantity = x.Quantity,
-                    StockQuantity = x.StockQuantity,
-                    OutletQuantity = x.OutletQuantity,
-                    ExternalReference = x.ExternalReference,
-                    MoreInfo = x.MoreInfo
-                }));
+            var basket = await _basketRepository.SaveAsync(token, language, id, items);
```

Comment:
**suggestion (non-blocking):** if id is not used out of SaveAsync method, you can use directly Guid.Parse(reqCookie)